plugin/src/main/java/org/wso2/carbon/uuf/maven/bean/ComponentConfig.java

```diff
@@ -168,25 +214,26 @@ public String getMode() {
          * Sets the mode of this binding.
          *
          * @param mode mode to be set
-         * @exception IllegalArgumentException if the {@code mode} is not {@link #MODE_PREPEND} or {@link #MODE_APPEND}
-         *                                     or {@link #MODE_OVERWRITE}
+         * @throws IllegalArgumentException if the {@code mode} is not {@link #MODE_PREPEND} or {@link #MODE_APPEND} or
+         *                                  {@link #MODE_OVERWRITE}
          * @see #MODE_PREPEND
          * @see #MODE_APPEND
          * @see #MODE_OVERWRITE
          */
         public void setMode(String mode) {
-            if (MODE_PREPEND.equals(mode) || MODE_APPEND.equals(mode) || MODE_OVERWRITE.equals(
-                    mode)) {
+            if (mode == null) {
+                this.mode = MODE_PREPEND; // default mode is prepend.
+            } else if (MODE_PREPEND.equals(mode) || MODE_APPEND.equals(mode) || MODE_OVERWRITE.equals(mode)) {
                 this.mode = mode;
             } else {
                 throw new IllegalArgumentException(
-                        "Binding mode should be either '" + MODE_PREPEND + "', '" + MODE_APPEND + "' or '" +
-                                MODE_OVERWRITE + "'. Instead found '" + mode + "'.");
+                        "Mode of a binding entry in the component's config should be either '" + MODE_PREPEND + "', '" +
```

Comment:
I think it looks better if we move the message to the previous line (Because the message will anyway take 2 lines)